website/common/script/ops/buy/buyMarketGear.js

```diff
@@ -31,6 +31,10 @@ export class BuyMarketGearOperation extends AbstractGoldItemOperation {
     let item = content.gear.flat[key];
     if (!item) throw new NotFound(errorMessage('itemNotFound', {key}));
 
+    if (item.klass && item.klass !== 'special' && item.klass !== user.stats.class) {
+      throw new NotAuthorized(this.i18n('cannotBuyItem'));
+    }
+
     this.canUserPurchase(user, item);
```

Comment:
shouldn't the check go into this function?

Reply:
which check function?  canUserPurchase() ?

Reply:
Yes, canUserPurchase

Reply:
Oh I guess I forgot this condition :( , but @BlkStormy thanks for fixing it :)

`canUserPurchase` is from `AbstractGoldItemOperation` so a check for `item.klass` isn't something that needs to be checked for every gold-purchasable-item.

You could either override `canUserPurchase` (with the call to `super.canUserPurchase`) in this class and add the check there - OR - extend the checks for `item.canOwn` cuase this already checked inside `canUserPurchase` https://github.com/HabitRPG/habitica/blob/develop/website/common/script/ops/buy/abstractBuyOperation.js#L143

Reply:
@negue I just added another commit - can you take a look?  Is this what you meant?